src/test/java/net/testudobank/tests/MvcControllerIntegTest.java

```diff
@@ -591,6 +596,201 @@ public void testReversalOfSimpleWithdraw() throws SQLException, ScriptException,
+
+    //Verify that customer2's overdraft balance decreased by $100.
+    assertEquals((CUSTOMER2_EXPECTED_OVERDRAFT_BALANCE - TRANSFER_AMOUNT_IN_PENNIES), (int)customer2Data.get("OverdraftBalance"));
+
+    //Check that transfer request goes through.
+    assertEquals("account_info", returnedPage);
+
+}
+
+
+/**
+ * This test will test a scenario where the sender sends the recipient (who currently has an overdraft balance) an amount
+ * that clears the recipient's overdraft balance and deposits the remainder. 
+ * 
+ * The sender will be initialized with $1000, and the recipient will have an overdraft balance of $100. Due to applied interest,
+ * the recipient will have an overdraft balance of $102. The sender will send $150, so the recipient's balance should reflect $48
+ * after the transfer.
+ * 
+ * @throws SQLException
+ * @throws ScriptException
+ */
+
+@Test
+public void testTransferPaysOverdraftAndDepositsRemainder() throws SQLException, ScriptException { 
+
+   //Initialize customer1 with a balance of $1000. Balance will be represented as pennies in DB.
+   double CUSTOMER1_BALANCE = 1000;
+   int CUSTOMER1_BALANCE_IN_PENNIES = MvcControllerIntegTestHelpers.convertDollarsToPennies(CUSTOMER1_BALANCE);
+   MvcControllerIntegTestHelpers.addCustomerToDB(dbDelegate, CUSTOMER1_ID, CUSTOMER1_PASSWORD, CUSTOMER1_FIRST_NAME, CUSTOMER1_LAST_NAME, CUSTOMER1_BALANCE_IN_PENNIES);
+
+   //Initialize customer2 with a balance of $100. Balance will be represented as pennies in DB. 
+   double CUSTOMER2_BALANCE = 100;
+   int CUSTOMER2_BALANCE_IN_PENNIES = MvcControllerIntegTestHelpers.convertDollarsToPennies(CUSTOMER2_BALANCE);
+   MvcControllerIntegTestHelpers.addCustomerToDB(dbDelegate, CUSTOMER2_ID, CUSTOMER2_PASSWORD, CUSTOMER2_FIRST_NAME, CUSTOMER2_LAST_NAME, CUSTOMER2_BALANCE_IN_PENNIES);
+
+   //Transfer $150 from sender's account to recipient's account.
+   int TRANSFER_AMOUNT = 150;
+   int TRANSFER_AMOUNT_IN_PENNIES = MvcControllerIntegTestHelpers.convertDollarsToPennies(TRANSFER_AMOUNT);
+
+   //Initializing users for the transfer
+   User CUSTOMER1 = new User();
+   User CUSTOMER2 = new User();
+   CUSTOMER1.setUsername(CUSTOMER1_ID);
+   CUSTOMER1.setPassword(CUSTOMER1_PASSWORD);
+
+   double CUSTOMER2_AMOUNT_TO_WITHDRAW = 200;
+   int CUSTOMER2_AMOUNT_TO_WITHDRAW_IN_PENNIES = MvcControllerIntegTestHelpers.convertDollarsToPennies(CUSTOMER2_AMOUNT_TO_WITHDRAW);
+   int CUSTOMER2_EXPECTED_OVERDRAFT_BALANCE = MvcControllerIntegTestHelpers.applyOverdraftInterest(CUSTOMER2_BALANCE_IN_PENNIES - CUSTOMER2_AMOUNT_TO_WITHDRAW_IN_PENNIES) * -1;
+   CUSTOMER2.setUsername(CUSTOMER2_ID);
+   CUSTOMER2.setPassword(CUSTOMER2_PASSWORD);
+   CUSTOMER2.setAmountToWithdraw(CUSTOMER2_AMOUNT_TO_WITHDRAW);
+
+   CUSTOMER1.setWhoToTransfer(CUSTOMER2_ID);
+   CUSTOMER1.setAmountToTransfer(TRANSFER_AMOUNT);
+
+   //Withdraw $200 from Customer2's account to cause an overdraft of $100.
+   controller.submitWithdraw(CUSTOMER2);
```

Comment:
Again, you can just initialize customer2 to have an overdraft balance of $102 without doing these withdraws.